### DISCUSSION TITLE
LSP Inlay Hints

### DISCUSSION THREAD

**Discussion by dpetka2001** [LazyVim/LazyVim#1016] [Q&A] [answered]
I have enabled `opts.inlay_hints.enabled = true` in my lspconfig spec and I also enabled inlay hints for the corresponding servers (for example for tsserver)
```lua
javascript = {
  inlayHints = {
    includeInlayParameterNameHints = "all",
    includeInlayParameterNameHintsWhenArgumentMatchesName = false,
    includeInlayFunctionParameterTypeHints = true,
    includeInlayVariableTypeHints = true,
    includeInlayPropertyDeclarationTypeHints = true,
    includeInlayFunctionLikeReturnTypeHints = true,
    includeInlayEnumMemberValueHints = true,
  },
},
```
and then in the buffer I execute `lua vim.lsp.buf.inlay_hint(0, true)` and get the following shown in the image below

![2023-06-30_12-31](https://github.com/LazyVim/LazyVim/assets/12776461/bc3f340a-a239-4bbe-9bdd-9edc47273ee2)

If I then execute `lua vim.lsp.buf.inlay_hint(0, false)` the hints right after the parameters/variables disappear and only the ones at the end of the line remain, which are the originals from the LSP server itself. I would much rather only have the first hints shown and not the ones at the end of the line, as it clutters the viewspace quite a bit for my taste at least. Is there a way to achieve this? Am I missing something in my spec that could accomplish this?

**Comment by folke** [comment]
I'm a bit confused with your question. inlay hints **are** the lsp ones. There is not other way to show inlay hints than using the lsp.

If you disable inlay hints and you can still see some, then I'm pretty sure that's a Neovim bug, since they're supposed to be all removed after disabling.

**Reply by folke** [reply, marked as answer]
Do you still have an inlay-hint plugin active as well maybe?

There's two plugins that provide that functionality, but since this is now in core, those are no longer needed and should be removed

**Reply by dpetka2001** [reply]
Yes it was `lvimuser/lsp-inlayhints.nvim`. Disabled that and all is well now. Thank you for pointing me to the right direction. I had completely forgotten about it.
